keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ApplicationController.kt

```diff
@@ -44,32 +44,24 @@ class ApplicationController(
   )
   fun get(
     @PathVariable("application") application: String,
-    @RequestParam("includeDetails", required = false, defaultValue = "false") includeDetails: Boolean
+    @RequestParam("includeDetails", required = false, defaultValue = "false") includeDetails: Boolean,
+    @RequestParam("entities", required = false, defaultValue = "resources") entities: List<String>
   ): Map<String, Any> {
-    if (includeDetails) {
-      var resources = repository.getSummaryByApplication(application)
-      resources = resources.map { summary ->
-        if (actuationPauser.resourceIsPaused(summary.id)) {
-          // we only update the status if the individual resource is paused,
-          // because the application pause is reflected in the response as a top level key.
-          summary.copy(status = PAUSED)
-        } else {
-          summary
+    return mutableMapOf<String, Any>(
+      "applicationPaused" to actuationPauser.applicationIsPaused(application),
+      "hasManagedResources" to applicationService.hasManagedResources(application)
+    ).also { results ->
+      if (includeDetails) {
+        entities.forEach { entity ->
+          results[entity] = when (entity) {
+            "resources" -> applicationService.getResourceSummariesFor(application)
+            "environments" -> applicationService.getEnvironmentSummariesFor(application)
+            "artifacts" -> applicationService.getArtifactSummariesFor(application)
+            else -> throw InvalidRequestException("Unknown entity type: $entity")
+          }
```

Comment:
This allows for flexible requests from the UI -- you can specify one or more "entity types" to return in the response. Defaults to `"resource"` as that was the original behavior.

Reply:
Would this be sent as `entities=resources,environments,artifacts`? I believe that's how we are sending lists of things elsewhere

Reply:
No... This would be sent like `entities=resources&entities=environments&entities=artifacts` which is how multi-value HTTP parameters are mapped in Spring (or any HTTP request library, I believe). I can change that to do the parsing of the single parameter if you prefer, I just thought I'd follow the standard...

Reply:
Actually, never mind. It looks like Spring handles both formats automatically! 👍

Reply:
Great! Thank you for confirming.